snuba/clusters/__init__.py

```diff
@@ -0,0 +1,54 @@
+from snuba import settings
+
+from snuba.clickhouse.native import ClickhousePool, NativeDriverReader
+from snuba.clickhouse.query import ClickhouseQuery
+from snuba.reader import Reader
+
+class Cluster:
```

Comment:
I think it would make sense to split this into two classes (both concrete):

1. A `Cluster` that exposes `get_reader`.
2. A `ClickhouseCluster` that subclasses `Cluster` and exposes the additional methods that return the ClickHouse clients.

I think in an idealized model we'd be able to only depend on `Cluster` and all of the ClickHouse implementation details would be abstracted away, but it in the mean time this would allow us to understand what is still dependent on the ClickHouse-specific functionality.

Reply:
ping

Reply:
Updated